src/components/like-button/like-button.tsx

```diff
@@ -0,0 +1,104 @@
+
+import cn from 'clsx';
+import { ComponentProps, useState } from 'react';
+
+import { Icon } from '../icon';
+import { Label } from '../typography/label';
+
+export interface ButtonProps {
+  /**
+   * Button variant
+   */
+  variant?: 'unliked' | 'liked' | 'likes';
+  /**
+   * likes counter
+   */
+  likes?: number;
+  /**
+   * Optional disable icon
+   */
+  disabled?: boolean;
+  /**
+   * Optional click handler
+   */
+  onClick?: () => ComponentProps<'button'>['onClick'];
+}
+
```

Comment:
ich würde auf die varianten verzichten, diese logik soll innerhalb der komponente geschehen, weil sonst muss man von aussen die ganze logik aufbauen wann geliked wird usw.   
eine idee wäre es mit folgenden props umzusetzen 
```ts
  isLikedByUser?: boolean;
  onLikeAdd?: () => Promise<void>;
  onLikeRemove?: () => Promise<void>;
```
`onClick` & `variant` kann dann entfernt werden.  

damit kann man dann folgende Logik in der Komponente umsetzen:

```ts
export const LikeButton = ({
  disabled = false,
  likes = 0,
  isLikedByUser,
  onLikeRemove,
  onLikeAdd,
}: ButtonProps) => {
  const [variant, setVariant] = useState<'unliked' | 'liked' | 'likes'>(likes > 0 ? 'likes' : 'unliked');
  const [isLiked, setIsLiked] = useState(isLikedByUser);

  const handleLike = async () => {
    if (isLiked) {
      setIsLiked(false);
      // todo: disable button while remove is in progress
      await onLikeRemove?.();
    } else {
      setIsLiked(true);
      // todo: disable button while animating
      setVariant('liked');
      await onLikeAdd?.();
      new Promise((resolve) =>
        setTimeout(() => {
          setVariant('likes');
          resolve(true);
        }, 600),
      );
    }
  };

....
....
```
so kann man es auch besser in storybook zeigen